### PR TITLE
[1775] Handle representation/document with a name length > 50 characters

### LINKED ISSUE #1775: Rename a representation/document with a name length > 50 characters is not properly handled
* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- latest
* [x] **I have specified my environment** : 
- All

### Stack trace

Server stack trace after the rename:

```
2023-03-01 09:37:39.625 ERROR 98777 --- [14-f65b9545938c] o.h.engine.jdbc.spi.SqlExceptionHelper   : ERROR: new row for relation "representation" violates check constraint "representation_label_length"
  Détail : Failing row contains (df0bc3ba-3685-4a04-8ddf-2467f05262ff, 46e24162-b280-4843-af14-f65b9545938c, 4518b809-169c-473a-ac7a-b0a7ed36cb5a, Averyveryveryverylongnamewithoutspacesazertyuiopaz, siriusComponents://representation?type=Diagram, 29228857-eb70-346d-ba1f-b6844b21a977, {"id":"df0bc3ba-3685-4a04-8ddf-2467f05262ff","kind":"siriusCompo...).
2023-03-01 09:37:39.627  INFO 98777 --- [14-f65b9545938c] o.h.e.j.b.internal.AbstractBatchImpl     : HHH000010: On release of batch it still contained JDBC statements
2023-03-01 09:37:39.653  WARN 98777 --- [14-f65b9545938c] o.e.s.c.c.e.EditingContextEventProcessor : could not execute statement; SQL [n/a]; constraint [representation_label_length]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement

org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [representation_label_length]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:276) ~[spring-orm-5.3.23.jar:5.3.23]
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:233) ~[spring-orm-5.3.23.jar:5.3.23]
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:566) ~[spring-orm-
	at ...
	at org.eclipse.sirius.web.services.representations.RepresentationService.save(RepresentationService.java:123) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.diagrams.handlers.RenameDiagramEventHandler.handle(RenameDiagramEventHandler.java:91) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.diagrams.DiagramEventProcessor.handle(DiagramEventProcessor.java:119) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.handleRepresentationInput(EditingContextEventProcessor.java:325) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.doHandle(EditingContextEventProcessor.java:245) ~[classes/:na]
	at ... org.eclipse.sirius.components.collaborative.trees.handlers.RenameTreeItemEventHandler.handle(RenameTreeItemEventHandler.java:110) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.trees.TreeEventProcessor.handle(TreeEventProcessor.java:118) ~[classes/:na]
	....
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:109) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:37) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:200) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.batch.internal.NonBatchingBatch.addToBatch(NonBatchingBatch.java:46) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3571) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.updateOrInsert(AbstractEntityPersister.java:3438) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3852) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.action.internal.EntityUpdateAction.execute(EntityUpdateAction.java:202) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:604) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.spi.ActionQueue.lambda$executeActions$1(ActionQueue.java:478) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[na:na]
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:475) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:344) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:40) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1407) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:489) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:3303) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2438) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:449) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:562) ~[spring-orm-5.3.23.jar:5.3.23]
	... 36 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: new row for relation "representation" violates check constraint "representation_label_length"
  Détail : Failing row contains (df0bc3ba-3685-4a04-8ddf-2467f05262ff, 46e24162-b280-4843-af14-f65b9545938c, 4518b809-169c-473a-ac7a-b0a7ed36cb5a, Averyveryveryverylongnamewithoutspacesazertyuiopaz, siriusComponents://representation?type=Diagram, 29228857-eb70-346d-ba1f-b6844b21a977, {"id":"df0bc3ba-3685-4a04-8ddf-2467f05262ff","kind":"siriusCompo...).
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675) ~[postgresql-42.3.7.jar:42.3.7]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365) ~[postgresql-42.3.7.jar:42.3.7]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:355) ~[postgresql-42.3.7.jar:42.3.7]
	... 58 common frames omitted
```

### Steps to reproduce

1. Use the Robot Flow model
2. Select the System item in the Explorer View
3. Create a new representation on the system element with name > 50 characters

### Expected behavior

The new name is taken into account or is not taken into account but an error message appears in the snackbar.
It is possible to rename any other tree item or any field in the details view after that.

### Actual behavior

The new name is not taken into account.
It is impossible to rename any other tree item or any field in the details view after that.

### PR DESCRIPTION
**Pull request by AxelRICHARD**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/1775

# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [x] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - https://github.com/eclipse-sirius/sirius-components/issues/1746[#1746] [forms] Rich-text editor toolbar on forms representations are not grayed when read-only
 - https://github.com/eclipse-sirius/sirius-components/issues/1744[#1744] [forms] Checkboxes on forms representations are not grayed when read-only
 - https://github.com/eclipse-sirius/sirius-components/issues/1639[#1639] [core] Support schemaLocation and extendedMetaData for JSON resource
+- https://github.com/eclipse-sirius/sirius-components/issues/1775[#1775] [core] Increase representation/document/project label length from 50 to 1024 characters maximum
```
Comment (pcdavid) [inline CHANGELOG.adoc]: Do these even make sense in the UI?
I agree that we should not impose some arbitrary small limit on the name's length, but I'm not sure the different places the representation name can appear in our UI can actually handle long names. At least the tabs for currently opened representation do not handle long names very well IMO.

File: packages/sirius-web/backend/sirius-web-persistence/src/main/resources/db/changelog/sirius-web-schema.sql [hunk 2]
```diff
@@ -16,7 +16,7 @@ CREATE TABLE idmapping (
 CREATE TABLE project (
     id uuid DEFAULT gen_random_uuid() NOT NULL,
     name text NOT NULL,
-    CONSTRAINT project_name_length CHECK (((char_length(name) > 0) AND (char_length(name) <= 50)))
+    CONSTRAINT project_name_length CHECK (((char_length(name) > 0) AND (char_length(name) <= 1024)))
```
Comment (pcdavid) [inline packages/sirius-web/backend/sirius-web-persistence/src/main/resources/db/changelog/sirius-web-schema.sql:19]: Inconsistent with `ProjectService.isValidProjectName(String)` and `isNameInvalid` in `RenameProjectModalMachine`. Also, as for the representations I'm not sure our current UI would react well to actually long project names (no need to go as far as 1024).

**Review by pcdavid — DISMISSED** [review]

File: CHANGELOG.adoc (hunk 1 above)
Comment (AxelRICHARD) [reply CHANGELOG.adoc]: For the tabs, a PR has been proposed recently: https://github.com/eclipse-sirius/sirius-components/pull/1774

File: packages/sirius-web/backend/sirius-web-persistence/src/main/resources/db/changelog/sirius-web-schema.sql (hunk 2 above)
Comment (AxelRICHARD) [reply packages/sirius-web/backend/sirius-web-persistence/src/main/resources/db/changelog/sirius-web-schema.sql:19]: The previous value was also inconsistent.
The only goal here is to limit the risk to fall to the SQL error raised in https://github.com/eclipse-sirius/sirius-components/issues/1775.

File: CHANGELOG.adoc (hunk 1 above)
Comment (AxelRICHARD) [reply CHANGELOG.adoc]: Long names in the UI have been handled in the last version of the PR

**Review by sbegaudeau — APPROVED** [review]
I'll merge it after the build

**Pull request merged**